File: laba3/task2/Golang/stack.go
package main
import "fmt"

type Stack struct {
	top *Node
}

func (this *Stack) createNodeSt(data string) *Node {
	return createNode(data)
}

func (this *Stack) popStack() {
	if this.top == nil {
		return
	}
	this.top = this.top.next
}

func (this *Stack) printStack() {
	curr := this.top
	for curr != nil {
		fmt.Print(curr.data, " ")
		curr = curr.next
	}
	fmt.Println()
}

func (this *Stack) pushStack(data string) {
	node := this.createNodeSt(data)
	if this.top == nil {
		this.top = node
		return
	}
	node.next = this.top
	this.top = node
}

func (this *Stack) getTop() *Node { return this.top }

File: laba3/task3/C++/stack.cpp
#include <iostream>
#include "structures.h"

using namespace std;

void Stack::popStack()
{
	if (this->top == nullptr) return;
	const Node* temp = this->top;
	this->top = this->top -> next;
	delete temp;
}


void Stack::printStack() const
{
    const Node* curr = this->top;
    while (curr != nullptr) {
        cout << curr->data << " ";
        curr = curr->next;
    }
    cout << endl;
}


void Stack::pushStack(const string& data){
	Node* node = createNodeSt(data);
	if (this->top == nullptr){
    	this->top = node;
    	return;
    }
	node->next = this->top;
	this->top = node;
}

